tests/src/3d/testqgs3drendering.cpp

```diff
@@ -1408,5 +1469,155 @@ void TestQgs3DRendering::testFilteredExtrudedPolygons()
 
 }
 
+void TestQgs3DRendering::testDepthBuffer()
+{
+  // =============================================
+  // =========== creating Qgs3DMapSettings
+  const QString dataDir( TEST_DATA_DIR );
+  QgsRasterLayer *layerDtm = new QgsRasterLayer( dataDir + "/3d/dtm.tif", "dtm", "gdal" );
+  QVERIFY( layerDtm->isValid() );
+  QgsProject project;
+  project.addMapLayer( layerDtm );
+
+  const QgsRectangle fullExtent = layerDtm->extent();
+
+  Qgs3DMapSettings mapSettings;
+  mapSettings.setCrs( project.crs() );
+  mapSettings.setExtent( fullExtent );
+  mapSettings.setLayers( {layerDtm} );
+
+  mapSettings.setTransformContext( project.transformContext() );
+  mapSettings.setPathResolver( project.pathResolver() );
+  mapSettings.setMapThemeCollection( project.mapThemeCollection() );
+
+  QgsDemTerrainGenerator *demTerrain = new QgsDemTerrainGenerator;
+  demTerrain->setLayer( layerDtm );
+  mapSettings.setTerrainGenerator( demTerrain );
+  mapSettings.setTerrainVerticalScale( 3 );
+
+  QgsPointLightSettings defaultPointLight;
+  defaultPointLight.setPosition( QgsVector3D( 0, 1000, 0 ) );
+  defaultPointLight.setConstantAttenuation( 0 );
+  mapSettings.setLightSources( {defaultPointLight.clone() } );
+  mapSettings.setOutputDpi( 92 );
+
+  // =========== creating Qgs3DMapScene
+  QPoint winSize = QPoint( 640, 480 ); // default window size
+  QPoint midPos = winSize / 2;
+
+  QgsOffscreen3DEngine engine;
+  engine.setSize( QSize( winSize.x(), winSize.y() ) );
+  Qgs3DMapScene *scene = new Qgs3DMapScene( mapSettings, &engine );
+  engine.setRootEntity( scene );
+
+  // =========== set camera position
+  scene->cameraController()->setLookingAtPoint( QVector3D( 0, 0, 0 ), 1500, 40.0, -10.0 );
+
+  // =============================================
+  // =========== TEST DEPTH
+  QgsCameraController4Test *testCam = static_cast<QgsCameraController4Test *>( scene->cameraController() );
+
+  // retrieve 3D depth image
+  QImage depthImage = Qgs3DUtils::captureSceneDepthBuffer( engine, scene );
+  QImage grayImage = convertDepthImageToGray16Image( depthImage );
+  QVERIFY( renderCheck( "depth_retrieve_image", grayImage, 550 ) );
+
+  // =========== TEST WHEEL ZOOM
+  QVector3D startPos = scene->cameraController()->camera()->position();
+  // set cameraController mouse pos to middle screen
+  QMouseEvent mouseEvent( QEvent::MouseButtonPress, QPointF( midPos.x(), midPos.y() ),
+                          Qt::MiddleButton, Qt::MiddleButton, Qt::NoModifier );
+  testCam->superOnMousePressed( new Qt3DInput::QMouseEvent( mouseEvent ) );
+
+  // First wheel action
+  QWheelEvent wheelEvent( midPos, midPos, QPoint(), QPoint( 0, 120 ),
+                          Qt::NoButton, Qt::NoModifier, Qt::NoScrollPhase,
+                          false, Qt::MouseEventSynthesizedByApplication );
+  testCam->superOnWheel( new Qt3DInput::QWheelEvent( wheelEvent ) );
+  qDebug() << "Checking depth after onWheel called" ;
```

Comment:
remove this line or change it to a proper message.

Reply:
done!